problems/Dynamic Programming/LIS.md

```diff
@@ -0,0 +1,56 @@
+---
+sidebar_position: 2
+---
+
+# Longest Increasing Subsequence
+
+We are given an array with `n` numbers: `a[0... n-1]` . The task is to find the longest, strictly increasing, subsequence in `a`.
+
+## Solution in `O(nlogn)` using dynamic programming and Binary search
+
+We will use the dynamic programming array `d[0 ... n-1]`. `d[i]` will be the element at which a subsequence of length `i` terminates. If there are multiple such sequences, then we take the one that ends in the smallest element.
+
+Initially we assume `d[0] = −∞` and for all other elements `d[i] = ∞`.
+
+We will gradually process the numbers, first `a[0]`, then `a[1]`, etc, and in each step maintain the array `d[]` so that it is up to date.
+
+After processing all the elements of a[] the length of the desired subsequence is the largest `l` with `d[l] < ∞`.
+
+### C++ Code
+
+```cpp
```

Comment:
Should we put this in a spoiler? Try this https://stackoverflow.com/questions/60827828/docusaurus-v2-collapsible-section-in-markdown